noir-projects/aztec-nr/aztec/src/utils/comparison.nr

```diff
@@ -16,137 +16,103 @@ global Comparator = ComparatorEnum {
     GTE: 6,
 };
 
-pub fn assert_comparison<let N: u32>(lhs: Field, operation: u8, rhs: Field, error_msg: str<N>) {
+pub fn compare(lhs: Field, operation: u8, rhs: Field) -> bool {
     // Values are computed ahead of time because circuits evaluate all branches
     let is_equal = lhs == rhs;
     let is_lt = lhs.lt(rhs);
 
     if (operation == Comparator.EQ) {
-        assert(is_equal, error_msg);
+        is_equal
     } else if (operation == Comparator.NEQ) {
-        assert(!is_equal, error_msg);
+        !is_equal
     } else if (operation == Comparator.LT) {
-        assert(is_lt, error_msg);
+        is_lt
     } else if (operation == Comparator.LTE) {
-        assert(is_lt | is_equal, error_msg);
+        is_lt | is_equal
     } else if (operation == Comparator.GT) {
-        assert(!is_lt & !is_equal, error_msg);
+        !is_lt & !is_equal
     } else if (operation == Comparator.GTE) {
-        assert(!is_lt, error_msg);
+        !is_lt
+    } else {
+        assert(false, "Invalid operation");
+        false // Noir would complain without boolean value here
     }
 }
 
 mod test {
-    use super::assert_comparison;
+    use super::compare;
     use super::Comparator;
 
     #[test]
-    fn test_assert_comparison_happy_path() {
+    fn test_compare() {
         let lhs = 10;
         let rhs = 10;
-        assert_comparison(lhs, Comparator.EQ, rhs, "Expected lhs to be equal to rhs");
+        let result = compare(lhs, Comparator.EQ, rhs);
+        assert(result, "Expected lhs to be equal to rhs");
```

Comment:
I feel like this is a tad too verbose, don't you find `assert(compare(10, Comparator.EQ, 10))` to be be easier to read?

Reply:
(and `assert(!compare)` for the failing cases)